lib/libfrr.c

```diff
@@ -307,6 +310,7 @@ void frr_init_vtydir(void)
 void frr_preinit(struct frr_daemon_info *daemon, int argc, char **argv)
 {
 	di = daemon;
+	frr_is_after_fork = false;
```

Comment:
is there any potential issue with this assert and test for "oddball" deamons like ldpd, who go their own way?

Reply:
Hrm, yes, though I'm not sure it matters.  If they go their own way, wouldn't they also sidestep the `frr_pthread` bits?

Reply:
Plopped on a commit to set the flag in lde/ldpe.